src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -470,41 +470,79 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
             }
         }
 
-        public class BooleanSupport : JavascriptConversionExtension
+        public class ConstSupport : JavascriptConversionExtension
         {
             public override void ConvertToJavascript(JavascriptConversionContext context)
             {
-                if (!(context.Node is ConstantExpression nodeAsConst) ||
-                    nodeAsConst.Type != typeof(bool))
+                if (!(context.Node is ConstantExpression nodeAsConst))
                     return;
 
-                context.PreventDefault();
                 var writer = context.GetWriter();
-                var val = (bool)nodeAsConst.Value ? "true" : "false";
-
                 using (writer.Operation(nodeAsConst))
                 {
-                    writer.Write(val);
-                }
-            }
-        }
+                    if (nodeAsConst.Value == null)
+                    {
+                        context.PreventDefault();
+                        writer.Write("null");
+                        return;
+                    }
 
-        public class CharSupport : JavascriptConversionExtension
-        {
-            public override void ConvertToJavascript(JavascriptConversionContext context)
-            {
-                if (!(context.Node is ConstantExpression nodeAsConst) ||
-                    nodeAsConst.Type != typeof(char))
-                    return;
+                    if (nodeAsConst.Type == typeof(bool))
+                    {
+                        context.PreventDefault();
 
-                context.PreventDefault();
-                var writer = context.GetWriter();
+                        var val = (bool)nodeAsConst.Value ? "true" : "false";
 
-                using (writer.Operation(nodeAsConst))
-                {
-                    writer.Write("\"");
-                    writer.Write(nodeAsConst.Value);
-                    writer.Write("\"");
+                        using (writer.Operation(nodeAsConst))
+                        {
+                            writer.Write(val);
+                        }
+
+                        return;
+                    }
+
+                    if (nodeAsConst.Type == typeof(char))
+                    {
+                        context.PreventDefault();
+
+                        writer.Write("\"");
+                        writer.Write(nodeAsConst.Value);
+                        writer.Write("\"");
+
+                        return;
+
+                    }
+
+                    if (nodeAsConst.Type.IsArray || LinqMethodsSupport.IsCollection(nodeAsConst.Type))
```

Comment:
A cool optimization at some point would be to build object property map instead of array when parameter is of type string array or HashSet<string>, this might depend on size though. looku["stringvalue"] is dictionary lookup compared to array search.

Reply:
I think we want to avoid being too smart here.
The user can use a Dictionary / HashSet, and we'll recognize that.